app/src/main/java/com/android/unio/model/event/EventViewModel.kt

```diff
@@ -66,9 +68,16 @@ constructor(
    * Updates the selected event in the ViewModel.
    *
    * @param eventId the ID of the event to select.
+   * @param loadPictures if true, loads the user pictures from firestore
    */
-  fun selectEvent(eventId: String) {
-    _selectedEvent.value = findEventById(eventId).also { it?.taggedAssociations?.requestAll() }
+  fun selectEvent(eventId: String, loadPictures: Boolean = false) {
+    _selectedEvent.value =
+        findEventById(eventId).also {
+          it?.taggedAssociations?.requestAll()
+          if (loadPictures) {
+            it?.eventPictures?.requestAll()
+          }
+        }
```

Comment:
Suggestion: might be better to write a `loadPictures` function, that you call when the composable is opened, e.g. in an LaunchedEffect as I suggested in the main comment of the review.